microsoft/testsuites/security/fips.py

```diff
@@ -1,14 +1,357 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
+import json
 import re
+from abc import ABC, abstractmethod
+from typing import Any, Dict
 
 from assertpy import assert_that
 
-from lisa import Logger, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
-from lisa.operating_system import CBLMariner
-from lisa.tools import Cat
-from lisa.util import LisaException, SkippedException, get_matched_str
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.features import Disk
+from lisa.operating_system import CBLMariner, OperatingSystem
+from lisa.sut_orchestrator.azure.common import METADATA_ENDPOINT
+from lisa.tools import Blkid, Cat, Curl, Sed
+from lisa.util import SkippedException, get_matched_str
+
+
+def assert_package_exists(os: OperatingSystem, package: str) -> None:
+    """
+    Assert that a package/tool is installed on the node.
+    """
+    assert_that(os.package_exists(package)).described_as(
+        f"Package {package} is not installed."
+    ).is_true()
+
+
+def assert_package_not_exists(os: OperatingSystem, package: str) -> None:
+    """
+    Assert that a package/tool is not installed on the node.
+    """
+    assert_that(os.package_exists(package)).described_as(
+        f"Package {package} is installed."
+    ).is_false()
+
+
+class AzlFips(ABC):
```

Comment:
It can be implemented by a tool. We have some virtual tools, they are not actually commands, but easy to be abtracted for reusing. In the tool, the `create` method can replace create_instance. This class can be called as "Fips", and derives a "AzLinuxFips/AzLinuxV2Fips,AzLinuxV3Fips" classes as needed.